src/Electron/electron.ts

```diff
@@ -344,3 +370,133 @@ export class ElectronContainer extends WebContainerBase {
         });
     }
 }
+
+export class ElectronWindowManager {
+    private ipc: NodeJS.EventEmitter;
+    private browserWindow: any;
+
+    private lastBounds: Map<number, Rectangle> = new Map(); // BrowserWindow.id -> Rectangle
+    private ignoredWindows: number[] = []; // Array of BrowserWindow.id
+
+    public constructor(ipc?: NodeJS.EventEmitter, browserWindow?: any) {
+        this.ipc = ipc || require("electron").ipcMain;
+        this.browserWindow = browserWindow || require("electron").BrowserWindow;
+
+        this.ipc.on("desktopJS.window-setname", (event: any, message: any) => {
```

Comment:
Should we not put these 'well known' names into a separate class as consts?

Reply:
@psmulovics PR updated.  Moved to a class InternalMessageType with public static readonly fields for the strings.